### DISCUSSION TITLE
Comments section for each image incorporated into viewer

### DISCUSSION THREAD

**Discussion by jbhanks** [openseadragon/openseadragon#2500] [Ideas]
I was thinking that it would be nice to have a way for users to leave/read comments on individual images displayed by OSD, and am wondering if anyone has already made such a thing. Ideally, I would want the comments stuff to be hidden unless the user clicks a button in the viewer.

In my preliminary research, I've found a few disucssions that could probably help me figure something out, but if something already exists, I'd rather start with that.

* https://stackoverflow.com/questions/60325746/how-do-you-add-captions-to-individual-tile-sources-in-an-array-with-openseadrago

* I asked ChatGPT, and it suggested just making a whole viewer for each image, which seems pretty silly to me. 

If nothing else exists, my thought is to make a plugin that addes a "comments" button, and when clicked it brings up the comments page for that image in a page overlaid on the viewer. The comments page brought up would be to use Ian's suggestion from the above thread:

```
var captions = [ <?php echo "$captions"; ?> ];

viewer.addHandler('page', function(event) {
  var caption = captions[event.page];
  // Update the caption element with the caption
});
```

Except instead of loading a caption, it loads the link for the comments page related to the image.

Anyway, I'm not really a JS person, so digging into Ajax and other JS libraries is slow going for me. Thus, any links to existing projects, suggestions, would be helpful if I were to undertake this.

**Comment by iangilman** [comment]
Sounds lovely! I'm not aware of another implementation of this. Your general plan sounds good :)
